src/eradius_server.erl

```diff
@@ -398,3 +416,51 @@ inc_discard_counter(malformed, NasProp, Ms) ->
 inc_discard_counter(_Reason, NasProp, Ms) ->
     eradius_metrics:update_nas_request(dropped, NasProp#nas_prop.metrics_info, Ms),
     eradius_counter:inc_counter(packetsDropped, NasProp).
+
+%% check if we can use persistent_term for config
+%% persistent term was added in OTP 21.2 but we can't
+%% check minor versions with macros so we're stuck waiting
+%% for OTP 22
+-ifdef(HAVE_PERSISTENT_TERM).
+
+request_counter_account_match_spec_compile() ->
```

Comment:
Looks like, this code can be refactored and move to macros:
```erlang
-define(REQ_ACC_MATCH_SPEC_COMPILE, ets:match_spec_compile(ets:fun2ms(fun
    ({#attribute{id = ?RStatus_Type}, ?RStatus_Type_Start})  -> {accounting_start, accountRequestsStart};
    ({#attribute{id = ?RStatus_Type}, ?RStatus_Type_Stop})   -> {accounting_stop, accountRequestsStop};
    ({#attribute{id = ?RStatus_Type}, ?RStatus_Type_Update}) -> {accounting_update, accountRequestsUpdate} end))).

-define(RESP_ACC_MATCH_SPEC_COMPILE, ets:match_spec_compile(ets:fun2ms(fun
    ({#attribute{id = ?RStatus_Type}, ?RStatus_Type_Start})  -> {accounting_start, accountResponsesStart};
    ({#attribute{id = ?RStatus_Type}, ?RStatus_Type_Stop})   -> {accounting_stop, accountResponsesStop};
    ({#attribute{id = ?RStatus_Type}, ?RStatus_Type_Update}) -> {accounting_update, accountResponsesUpdate} end))).
```

Reply:
that would rerun ets:match_spec_compile every time the macro is used, i.e. every time inc_response_counter_accounting is executed. ms_transform.hrl will only optimize fun2ms, the match_spec_compile. By putting the result of match_spec_compile into persistent term, the compile is executed only once for the first invocation.

Reply:
Thanks.